### PR TITLE
Oy2 18587 - Update measures to use ids for titles(descriptions)

### PR DESCRIPTION
**Pull request by jdinh8124**
## Purpose
Updated measure IDs to use year/type/measure to point to a front-end static description list

#### Linked Issues to Close
https://qmacbis.atlassian.net/jira/software/c/projects/OY2/boards/208?modal=detail&selectedIssue=OY2-18587&assignee=616db787702bd0006a4cfda8

#### Pull Request Creator Checklist

- [x] This PR has an associated issue or issues.
- [x] The associated issue(s) are linked above.
- [x] This PR meets all acceptance criteria for those issues.
- [x] This PR and linked issue(s) are adequately documented
- [x] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.
- [x] Someone has been assigned this PR.
- [x] At least one person has been marked as reviewer on this PR.

#### Pull Request Reviewer/Assignee Checklist

- [ ] This PR has an associated issue or issues.
- [ ] The associated issue(s) are linked above.
- [ ] This PR meets all acceptance criteria for those issues.
- [ ] This PR and linked issue(s) are adequately documented
- [ ] This PR and linked issues(s) are a complete description of the changeset; an individual or team should be able to understand the issue(s) and changes by reading through this PR and it's links, with no further interaction.

### CODE REVIEW

**Comment by benjaminpaige** [comment]
so i think the only thing we are using the measuresList for now is just the descriptions in the coreset list & measureWrapper. So i was thinking it would change from being a list to an object , for ex:

const measureDescriptions = {
2021: {
"AMMAD":  "Antidepressant Medication Management",
....
}
}

and then we could check to see if a description exists in the measureDescriptions object first, if not use the one that is stored in the db (this would likely only be applicable to custom hh measures. )

thoughts?

File: services/app-api/handlers/coreSets/create.ts [hunk 1]
```diff
@@ -90,7 +91,7 @@ const createDependentMeasures = async (
         status: measure.autocompleteOnCreation
           ? Types.MeasureStatus.COMPLETE
           : Types.MeasureStatus.INCOMPLETE,
-        description: measure.description,
+        description,
```
Comment (benjaminpaige) [inline services/app-api/handlers/coreSets/create.ts:94]: Im not sure we even need to set description at all here

File: services/app-api/handlers/coreSets/create.ts [hunk 2]
```diff
@@ -76,7 +76,7 @@ const createDependentMeasures = async (
     // The State Year and ID are all part of the path
     const measureId = measure["measure"];
     // Dynamo only accepts one row as a key, so we are using a combination for the dynamoKey
-    const dynamoKey = `${state}${year}${coreSet}${measureId}`;
+    const dynamoKey = `${state}/${year}/${coreSet}/${measureId}`;
```
Comment (jdinh8124) [inline services/app-api/handlers/coreSets/create.ts:79]: How do we feel about this?

File: services/ui-src/src/components/MeasureWrapper/index.tsx [hunk 3]
```diff
@@ -323,6 +324,16 @@ export const MeasureWrapper = ({
     return null;
   }
 
+  const formatTitle = (measureKey: string, customDescription?: string) => {
+    // Destruct out only the year and id
+    const [, itemYear, , itemMeasureId] = measureKey?.split("/");
```
Comment (benjaminpaige) [inline services/ui-src/src/components/MeasureWrapper/index.tsx:329]: same as below, you should have the year and measureId from the Item, so just pass the item in this formatTitle function. use the year and measureId from the item to check if it exists in the measuresDescription object, if not, use the description from the item, which should exist.

File: services/app-api/handlers/coreSets/create.ts (hunk 2 above)
Comment (benjaminpaige) [reply services/app-api/handlers/coreSets/create.ts:79]: im all for cleaning up the key like this. as long as it doesnt have any residual effects on existing db items

**Comment by benjaminpaige** [comment]
that feels a lot cleaner 👍

**Review by benjaminpaige — APPROVED** [review]

**Review by abelzer212 — APPROVED** [review]

**Pull request merged**